### PR TITLE
Adding an example on near-dup

### PR DESCRIPTION
**Pull request by sayakpaul**
In this example, we discuss some of the most widely used techniques (LSH, Random Projection, etc.) for building near-duplication detection systems for information retrieval. We also optimize the performance (on GPUs) of our embedding model by quantizing it using the TensorRT framework. 

A Colab Notebook is available [here](https://colab.research.google.com/github/sayakpaul/near-dup-parser/blob/main/bit-near-dup-parser.ipynb).

### CODE REVIEW

File: examples/keras_recipes/near_dup_parser.py [hunk 1]
```diff
@@ -0,0 +1,532 @@
...
+
+The underlying model is a BiT-ResNet (proposed in
+[Big Transfer (BiT): General Visual Representation Learning](https://arxiv.org/abs/1912.11370)).
+BiT-ResNet family of models are known to provide excellent transfer performance across
+a wide variety of different downstream tasks. 
+"""
+
+"""shell
+!gdown --id 19VxuBZyn9BUedwRCZRsXQ0-1HFbF7x3u
+!tar xf flower_model_bit_0.96875.gz
```
Comment (sayakpaul) [inline examples/keras_recipes/near_dup_parser.py:92]: Will replace this link after hosting the model as a GitHub release artifact.

File: examples/keras_recipes/near_dup_parser.py [hunk 2]
```diff
@@ -0,0 +1,538 @@
...
+lsh_builder.train(training_files)
+
+
+"""
+At the time of writing the wall time was 54.1 seconds on a Tesla T4 GPU. This timing may
+vary based on the GPU you are using.
+"""
+
+"""
+## Optimizing the model with TensorRT
```
Comment (fchollet) [inline examples/keras_recipes/near_dup_parser.py:329]: That's very useful, it's not something that's covered often.

File: examples/keras_recipes/near_dup_parser.py [hunk 3]
```diff
@@ -0,0 +1,538 @@
+"""
+Title: Near duplicate image parsing
```
Comment (fchollet) [inline examples/keras_recipes/near_dup_parser.py:2]: parsing -> search

**Review by fchollet — COMMENTED** [review]
Thanks for the PR! This is a great topic, and the mention of TensorRT is a great touch.

File: examples/keras_recipes/near_dup_parser.py [hunk 4]
```diff
@@ -0,0 +1,538 @@
...
+representations of the images. For this purpose, we will be branching the above image
+classifier such that it becomes apt for our purpose.
+"""
+
+embedding_model = tf.keras.Sequential(
+    [
+        tf.keras.layers.InputLayer((IMAGE_SIZE, IMAGE_SIZE, 3)),
+        tf.keras.layers.experimental.preprocessing.Rescaling(scale=1.0 / 255),
+        bit_model.layers[1],
+        tf.keras.layers.Lambda(tf.math.l2_normalize),
```
Comment (fchollet) [inline examples/keras_recipes/near_dup_parser.py:111]: Here, you can use a `Normalization(mean=0, variance=1)` layer.  Using `l2_normalize` without specifying an axis will normalize the entire batch, which is not what you intend.

Comment (sayakpaul) [reply examples/keras_recipes/near_dup_parser.py:111]: Thanks for spotting this!

**Comment by sayakpaul** [comment]
@fchollet thank you for all the feedback and comments. I have addressed them all.

**Review by fchollet — APPROVED** [review]
Awesome, thanks for the update! I added some minor copyedits. Once pulled, you can add the generated files.

**Comment by sayakpaul** [comment]
@fchollet ready for you to take over.

**Comment by fchollet** [comment]
Thank you for the great contribution!
